### ISSUE TITLE
allow --endpoint-url option to work with S3 compatible services

### ISSUE THREAD

**Issue by rmanus** [generalui/s3p#34]
It is currently not possible to use s3p with non AWS S3 compatible services.

**Comment by lincdog** [comment]
I also had this problem, I added this at line 59 of `s3p/source/S3Parallel/Lib/S3.js` and it worked. 
In my case I'm using Wasabi instead of AWS S3 but you can fill in your desired endpoint of course.
```javascript
this.wasabiEndpoint = new (require("aws-sdk").Endpoint)("https://s3.us-west-1.wasabisys.com")                                                                         this.awsSdkS3 = new (require("aws-sdk").S3)({endpoint: this.wasabiEndpoint}); 
```

**Comment by shanebdavis** [comment]
Good idea! I'll look into adding it. I think just `--endpoint [url]` will suffice.

**Comment by ihulsbus** [comment]
This would be a really useful feature for us as it enables us to use this in other S3 implementations such as Ceph. Any idea if and possibly when this could be available?

**Comment by jamongeon1** [comment]
Hi, is there any update if the --endpoint flag will be added?

**Comment by AriHuttunen** [comment]
I would also like to have this.

**Comment by guilhermesupe** [comment]
> I also had this problem, I added this at line 59 of `s3p/source/S3Parallel/Lib/S3.js` and it worked. In my case I'm using Wasabi instead of AWS S3 but you can fill in your desired endpoint of course.
> 
> ```js
> this.wasabiEndpoint = new (require("aws-sdk").Endpoint)("https://s3.us-west-1.wasabisys.com");
> this.awsSdkS3 = new (require("aws-sdk").S3)({endpoint: this.wasabiEndpoint}); 
> ```

in which function should I add this? Sorry I don't know much of JS but I really need to use this on wasabi as well, I think line 59 is not the same anymore

**Comment by tlan16** [comment]
Not supporting custom api endpoint kill any usage of self-hosted aws s3, e.g. localstack.

**Comment by jplitza** [comment]
While not exactly comfortable, you can put the following in `~/.aws/config`, which will be honored by s3p:
```ini
[default]
ignore_configure_endpoint_urls = true
endpoint_url = https://s3.example.com
```
